test/units/galaxy/test_collection_install.py

```diff
@@ -434,6 +434,58 @@ def test_build_requirement_from_tar_invalid_manifest(tmp_path_factory):
         Requirement.from_requirement_dict({'name': to_text(tar_path)}, concrete_artifact_cm)
 
 
+def test_build_requirement_from_tar_unsupported_collection_keys(collection_artifact):
+    tmp_path = os.path.join(os.path.split(collection_artifact[1])[0], b'temp')
+    concrete_artifact_cm = collection.concrete_artifact_manager.ConcreteArtifactsManager(tmp_path, validate_certs=False)
+    expected = (
+        r'The following {collection_name!s} collection requirement entry keys are not '
+        r'valid: invalid_key, key'.format(collection_name=to_text(collection_artifact[1]))
+    )
+    with pytest.raises(AnsibleError, match=expected):
+        Requirement.from_requirement_dict({'name': to_text(collection_artifact[1]), 'invalid_key': 'foo', 'key': 'bar'}, concrete_artifact_cm)
+
+
+def test_build_requirement_from_tar_invalid_name_with_source(collection_artifact):
```

Comment:
Minor nit: all the new tests following and including this one should not include `from_tar` in the name, since the `name` is not a path to a tarfile and the `type` is not explicitly specified as `file`.

```suggestion
def test_build_requirement_invalid_name_with_source(collection_artifact):
```